Engine/SubscriptionDataReader.cs

```diff
@@ -554,23 +558,29 @@ private SubscriptionStreamReader GetReader(string source)
                     // check if this is not a local uri then download it to the local cache
                     if (uri.IsAbsoluteUri && !uri.IsLoopback)
                     {
-                        try
+                        //Skip download if the file is already in cache
+                        if (!File.Exists(location))
```

Comment:
@bdilber I think we'll need to revert the change in behavior here since there's no way to know that the file we have is the correct file, or just has the same name. Also, the issue of holding possibly stale data comes into play.